### ISSUE TITLE
__attribute__( ( always_inline ) ) gives error in tx_port.h file 

### ISSUE THREAD

**Issue by kaushaleinfo** [azure-rtos/threadx#225]
**Describe the bug**
while compiling threadX project in IAR IDE V 6.5 gives error __attribute__( ( always_inline ) ) in tx_port.h file. 

Please also mention any information which could help others to understand
the problem you're facing:
- What target device are you using? --> Stm32f205
- Which version of Azure RTOS?        --> 6.2.0
- What toolchain and environment?  --> IAR IDE version 6.2.0
- What have you tried to diagnose or workaround this issue?  -> if I removed __attribute__( ( always_inline ) ) then it is compiling successfully 

**To Reproduce**
1. Build my project in IAR Workbench.
1. See error

**Expected behavior**
It shouldn't give error 

**Impact**
Not able to compile threadX project 

**Logs and console output**
__attribute__( ( always_inline ) ) static inline unsigned int __disable_interrupts(void)
{
unsigned int int_posture;

    int_posture = __get_interrupt_posture();

#ifdef TX_PORT_USE_BASEPRI
    __set_basepri_value(TX_PORT_BASEPRI);
#else
    __asm__ volatile ("CPSID i" : : : "memory");
#endif
    return(int_posture);
}

**Additional context**

**Comment by goldscott** [comment]
Perhaps this is due to a very old version of IAR. Can you please update to a newer version of EWARM?

**Comment by kaushaleinfo** [comment]
Yes It is working in latest version of EWARM but it is necessary to work with IAR IDE 6.5 only.
Please give us some solutions on that.

Thanks & Regards,
Kaushal Panchal
From: Scott ***@***.***>
Sent: 20 January 2023 01:25 AM
To: azure-rtos/threadx ***@***.***>
Cc: Kaushal Panchal ***@***.***>; Author ***@***.***>
Subject: [External] Re: [azure-rtos/threadx] __attribute__( ( always_inline ) ) gives error in tx_port.h file (Issue #225)


CAUTION: This email originated from outside of the organization. This message might not be safe, use caution in opening it. If in doubt, do not open the attachment nor links in the message.


Perhaps this is due to a very old version of IAR. Can you please update to a newer version of EWARM?

-

**Comment by goldscott** [comment]
It sounds like you've already found a solution - remove the attribute.

**Comment by kaushaleinfo** [comment]
Is it okay to remove the attribute?
I mean does it give any negative impact on performance or functionality?

From: Scott ***@***.***>
Sent: 20 January 2023 11:07 PM
To: azure-rtos/threadx ***@***.***>
Cc: Kaushal Panchal ***@***.***>; Author ***@***.***>
Subject: [External] Re: [azure-rtos/threadx] __attribute__( ( always_inline ) ) gives error in tx_port.h file (Issue #225)


CAUTION: This email originated from outside of the organization. This message might not be safe, use caution in opening it. If in doubt, do not open the attachment nor links in the message.


It sounds like you've already found a solution - remove the attribute.

-

**Comment by kaushaleinfo** [comment]
Is it okay to remove the attribute?
I mean does it give any negative impact on performance or functionality?

**Comment by kaushaleinfo** [comment]
Is it okay to remove the attribute?
I mean does it give any negative impact on performance or functionality?

**Comment by goldscott** [comment]
The compiler is probably smart enough to inline the function call anyway, though it likely depends on your optimization settings. To be absolutely sure, you should look at the generated assembly code to see if it in-lines the functions or not.
